internal/cloud/realm/auth.go

```diff
@@ -12,35 +12,80 @@ type loginPayload struct {
 	PrivateAPIKey string `json:"apiKey"`
 }
 
-// AuthResponse is the Realm login response
-type AuthResponse struct {
+// Session is the Realm login response
+type Session struct {
 	AccessToken  string `json:"access_token"`
 	RefreshToken string `json:"refresh_token"`
 }
 
-func (c *client) Authenticate(publicAPIKey, privateAPIKey string) (AuthResponse, error) {
-	path := adminAPI + "/auth/providers/mongodb-cloud/login"
+// UserProfile contains all of the roles associated with a user
+type UserProfile struct {
+	Roles []Role `json:"roles"`
+}
+
+// Role contains a GrouID field that maps to a project ID
+type Role struct {
+	GroupID string `json:"group_id"`
+}
+
+// AllGroupIDs returns all available group ids for a given user
+func (pd UserProfile) AllGroupIDs() []string {
+	groupIDs := []string{}
+	set := make(map[string]bool)
```

Comment:
I prefer using `struct{}` instead of `bool` here as a value for a set map

also, lets just keep the initializations to 

```go
var arr []string
set := map[string]struct{}{}
```